jsHelper/spicetifyWrapper.js

```diff
@@ -303,6 +303,86 @@ window.Spicetify = {
 	Platform: {}
 };
 
+(function addProxyCosmos() {
+	if (!Spicetify.Player.origin?._cosmos) {
+		setTimeout(addProxyCosmos, 50);
+		return;
+	}
+
+	const corsProxyURL = "https://cors-proxy.ririxi.workers.dev/";
+	const allowedMethodsMap = {
+		get: "get",
+		post: "post",
+		del: "delete",
+		put: "put",
+		patch: "patch"
+	};
+	const allowedMethodsSet = new Set(Object.keys(allowedMethodsMap));
+	const internalEndpoints = new Set(["sp:", "wg:"]);
+
+	const handler = {
+		// biome-ignore lint/complexity/useArrowFunction: <explanation>
+		get: function (target, prop, receiver) {
+			const internalFetch = Reflect.get(target, prop, receiver);
+
+			if (typeof internalFetch !== "function" || !allowedMethodsSet.has(prop) || Spicetify.Platform.version < "1.2.31") return internalFetch;
+
+			// biome-ignore lint/complexity/useArrowFunction: <explanation>
+			return async function (url, paramsOrBody) {
+				const urlObj = new URL(url);
+
+				const isWebAPI = urlObj.hostname === "api.spotify.com";
+				const isSpClientAPI = urlObj.hostname.includes(".spotify.com") && urlObj.hostname.includes("spclient");
+				const isInternalURL = internalEndpoints.has(urlObj.protocol);
+				// biome-ignore lint/style/noArguments: <explanation>
+				if (isInternalURL) return internalFetch.apply(this, arguments);
+
+				const shouldUseCORSProxy = !isWebAPI && !isSpClientAPI && !isInternalURL;
+
+				const method = allowedMethodsMap[prop.toLowerCase()];
+				const headers = {
+					"Content-Type": "application/json"
+				};
+
+				const options = {
+					method,
+					headers,
+					timeout: 1000 * 15
+				};
+
+				let finalURL = urlObj.toString();
+				if (paramsOrBody) {
+					if (method === "get") {
+						const params = new URLSearchParams(paramsOrBody);
+						finalURL += `?${params.toString()}`;
+					} else options.body = JSON.stringify(paramsOrBody);
+				}
+				if (shouldUseCORSProxy) finalURL = `${corsProxyURL}${finalURL}`;
+
+				const Authorization = `Bearer ${Spicetify.Platform.AuthorizationAPI._state.token.accessToken}`;
+				let injectedHeaders = {};
+				if (isWebAPI) injectedHeaders = { Authorization };
+				if (isSpClientAPI) {
+					injectedHeaders = {
+						Authorization,
+						"Spotify-App-Version": Spicetify.Platform.version,
+						"App-Platform": Spicetify.Platform.PlatformData.app_platform
+					};
+				}
+				Object.assign(options.headers, injectedHeaders);
+
+				try {
+					return fetch(finalURL, options).then(res => res.json());
```

Comment:
Doing `res.json()` on the response of a POST request (at least for adding an item to the queue), results in the following error which triggers my error handling, while the POST request itself was successful:
`SyntaxError: Unexpected end of JSON input`

Reply:
example of the request so i can test it on my end?

Reply:
await Spicetify.CosmosAsync.post(`https://api.spotify.com/v1/me/player/queue?uri=${uri}`);

Insert a uri yourself. I'm on my phone so I don't have one.

Reply:
Also use PlatformAPI for that tho... `Spicetify.Platform.PlayerAPI.addToQueue`